cpp/open3d/t/geometry/PointCloud.cpp

```diff
@@ -395,14 +401,21 @@ PointCloud PointCloud::FarthestPointDownSample(size_t num_samples) const {
         utility::LogError(
                 "Illegal number of samples: {}, must <= point size: {}",
                 num_samples, num_points);
+    } else if (start_index >= size_t(num_points)) {
+        utility::LogError("Illegal start index: {}, must <= point size: {}",
+                          start_index, num_points);
+    } else if (start_index >
+               static_cast<size_t>(std::numeric_limits<int64_t>::max())) {
```

Comment:
This check ensures that the `static_cast` down below does not result in undefined behavior. Is there a better way to do this check?

Also, not fully sure why `int64_t` is being used in the first place in this function

Reply:
The first check is sufficient because num_points is not larger than the max of in64_t.

int64_t is the default type for indices and shapes for open3d.core.Tensor.